abci/client/socket_client.go

```diff
@@ -99,7 +94,10 @@ func (cli *socketClient) OnStop() {
 		cli.conn.Close()
 	}
 
-	cli.drainQueue()
+	// this timeout is arbitrary.
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
Can we avoid having to wait an arbitrary amount here by closing the channel, so that the for loop can just wait for that instead?

```go
for rr := range cli.reqQueue {
   rr.Done()
}
```

Since we know we're shutting down anyway, I think that should be OK, though I would not be astonished to learn we are still issuing requests during shutdown 👀

Reply:
I do not feel like I have a good answer for this question. 

I think we're only ever going to call this once during actual shut down, this is really mostly just running in a signal handler.